clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -181,6 +215,15 @@ public synchronized void update(Cluster cluster, long now) {
     }
 
     /**
+     * If expiry is enabled, remove the unknown topic from the metadata refresh set to
+     * avoid further metadata requests for deleted topics which are not in use.
+     */
+    public synchronized void handleUnknownTopic(String topic) {
```

Comment:
Thanks @rajinisivaram. I think your change looks good. Some comments:

* `Metadata` isn't public (confusingly public classes are the ones we publish javadoc for)
* If it makes sense to call `update` without unknown topics during initialisation from streams and connect, then it makes sense for the overload to stay. It would be good to reference the other overload in the documentation with a short explanation.
* Maybe we should rename `handleFailedUpdate` back to `failedUpdate`. I somehow missed that we already had a method that we could modify for our purposes.